File: codice/matlab/rkt_dyn.m
function dX=rkt_dyn(t,X)
global g A cD Ap Isp th cDp rho tb td tT

x=X(1);
y=X(2);
vx=X(3);
vy=X(4);
m=X(5);

gamma=atan2(vy,vx);
v=sqrt(vx^2+vy^2);

if t<=tb+td
    D=0.5*rho*v^2*A*cD;
else
    D=0.5*rho*v^2*Ap*cDp; % exp opening
end

if t<=tb
    Ta=interp1(tT(:,1),tT(:,2),t);
    dm=-Ta/g/Isp;
else
    Ta=0;
    dm=0;
end

ax=Ta*cos(th)/m-D/m*cos(gamma);
ay=Ta*sin(th)/m-D/m*sin(gamma)-g;

dX(1,1)=vx;
dX(2,1)=vy;
dX(3,1)=ax;
dX(4,1)=ay;
dX(5,1)=dm;

File: codice/python/originale_tradotto/rkt_dyn.py
import numpy as np
import scipy.integrate as solve_ivp
import math, scipy

import rocketmodel as rk

# APPUNTI:
# gamma è l'angolo di inclinazione con l'orizzontale
# primva versione: copia brainless del codice originale
#
#
#
#

# vars:

def simulate_launch(t, vars):
    (m, x, y) = vars[0]
    (m_dot, x_dot, y_dot) = vars[1]
    
    gamma = math.atan2(y_dot, x_dot)
    v = math.sqrt(x_dot * x_dot + y_dot * y_dot)

    if t <= rk.tb + rk.td:
        D = 0.5 * rk.rho * v * v * rk.A * rk.cD
    else:
        D = 0.5 * rk.rho * v * v * rk.Ap * rk.cDp # exp opening (qualsiasi cosa sia)
    
    if t <= rk.tb:
        Ta = f(scipy.interp1d(rk.tT[:,1], rk.tT[:,2], kind = 'cubic'), t)
        rk.dm -=Ta/rk.g/rk.Isp
    else:
        Ta = 0
        rk.dm = 0

    ax = Ta * math.cos(rk.th) / m - D / m * math.cos(gamma);
    ay = Ta * math.sin(rk.th) / m - D / m * math.sin(gamma) - g;
    
    return [vars[1], (0., ax, ay)]